File: api/Models/Adoption.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace apifmu.Models
{
    public class Adoption
    {
        [Key]
        public int Id { get; set; }

        public char Situation { get; set; }

        public string DisapprovalReason { get; set; }

        [ForeignKey("Ong")]
        public int OngId { get; set; }

        public Ong Ong { get; set; }

        [ForeignKey("Pet")]
        public int PetId { get; set; }

        public Pet Pet { get; set; }

        [ForeignKey("User")]
        public int UserId { get; set; }

        public User User { get; set; }
    }
}

File: ui/src/models/adoption.ts
import { defaultOng, Ong } from './ong'
import { defaultPet, Pet } from './pet'
import { defaultUser, User } from './user'

export interface Adoption {
  id: number
  situation: 'A' | 'R' | 'W'
  disapprovalReason: string
  ongId: number
  ong: Ong
  petId: number
  pet: Pet
  userId: number
  user: User
}

export const defaultAdoption = (): Adoption => {
  return {
    id: 0,
    situation: 'W',
    disapprovalReason: '',
    ongId: 0,
    ong: defaultOng(),
    petId: 0,
    pet: defaultPet(),
    userId: 0,
    user: defaultUser(),
  }
}
